src/main/java/edu/nicholaidudakiwwarrick/advancedjava/app/StockQuoteApplication.java

```diff
@@ -0,0 +1,122 @@
+package edu.nicholaidudakiwwarrick.advancedjava.app;
+
+import edu.nicholaidudakiwwarrick.advancedjava.model.StockQuote;
+import edu.nicholaidudakiwwarrick.advancedjava.services.ServiceFactory;
+import edu.nicholaidudakiwwarrick.advancedjava.services.ServiceType;
+import edu.nicholaidudakiwwarrick.advancedjava.services.StockService;
+import edu.nicholaidudakiwwarrick.advancedjava.services.StockServiceException;
+import edu.nicholaidudakiwwarrick.advancedjava.util.IntervalEnum;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormatter;
+
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+/**
+ * Basic class with a Main method to create StockService instances
+ */
+public class StockQuoteApplication {
+    /**
+     * @param args an array that contains elements:
+     *             <code>String</code> representation of stock symbol, start date, and end date
+     */
+    public static void main(@NotNull String[] args) {
+
+        StockService stockService = null;
+
+        // create a StockService instance using the ServiceFactory static factory method
+        try {
+            stockService = ServiceFactory.getStockServiceInstance(ServiceType.BASIC);
+        } catch (StockServiceException e) {
+            e.getMessage();
+        }
+
+        // Retrieve date formatter and parse the 2nd and 3rd date args
+        DateTimeFormatter formatter = StockQuote.getDateFormatter();
+        DateTime startDate = DateTime.parse(args[1], formatter);
+        DateTime endDate = DateTime.parse(args[2], formatter);
+
+        System.out.println("Basic Stock Service Quotes");
+        System.out.println("****************************************");
+        try {
+            // Print a single stock quote to the console
+            System.out.println("Print a single stock quote");
+            System.out.println(stockService.getQuote(args[0], startDate).toString());
+        } catch (StockServiceException e) {
+            e.getMessage();
+        }
+
+        try {
+            // Print multiple stock quotes that fall between both date ranges
+            System.out.println("Print multiple stock quotes for the given date range");
+            List<StockQuote> stockList = stockService.getQuote(args[0], startDate, endDate);
+
+            for (StockQuote quote : stockList) {
+                System.out.println(quote.toString());
+            }
+        } catch (StockServiceException e) {
+            e.getMessage();
+        }
+        // Print multiple stock quotes between the period at the given interval
+        System.out.println("Print multiple stock quotes for the given date range at the given interval");
+        if(args[3].toUpperCase().equals("HOURLY")) {
+            List<StockQuote> intervalStockList =
```

Comment:
Whenever you write code that addresses a specific index value like [3] you risk an index out of bounds exception.  So you should be really carefully doing this (actually the best practice is to void it where possible) . 

One way to do this is to use a library to help parse the arguments like Args4j